numpy/ma/core.py

```diff
@@ -3648,6 +3659,13 @@ def fill_value(self):
     @fill_value.setter
     def fill_value(self, value=None):
         target = _check_fill_value(value, self.dtype)
+        if not target.ndim == 0:
+            warnings.warn(
+                "Non-scalar arrays for the fill value are deprecated. Use "
+                "arrays with scalar values instead. The filled function "
+                "still supports any array as `fill_value`.",
+                DeprecationWarning, stacklevel=2)
+
```

Comment:
Needs a date and version comment so we know when we can turn the deprecation into an error